CMakeLists.txt

```diff
@@ -51,12 +51,20 @@ FIND_PACKAGE(PkgConfig)
 FIND_PACKAGE(Threads REQUIRED)
 
 IF ((CMAKE_SYSTEM_NAME STREQUAL "Darwin") AND NOT (DEFINED OPENSSL_ROOT_DIR OR DEFINED ENV{OPENSSL_ROOT_DIR}))
-    MESSAGE(STATUS "*************************************************************************\n"
-                "   * Setting OPENSSL_ROOT_DIR to /usr/local/opt/openssl. On macOS, OpenSSL *\n"
-                "   * should be installed using homebrew or OPENSSL_ROOT_DIR must be set to *\n"
-                "   * the path that has OpenSSL installed.                                  *\n"
-                "   *************************************************************************")
-    SET(OPENSSL_ROOT_DIR "/usr/local/opt/openssl")
+    EXECUTE_PROCESS(
+        COMMAND brew --prefix openssl
+        OUTPUT_VARIABLE OPENSSL_ROOT_DIR
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+        ERROR_QUIET
+    )
+    IF (NOT EXISTS ${OPENSSL_ROOT_DIR})
+        MESSAGE(STATUS "*************************************************************************\n"
+                    "   * Setting OPENSSL_ROOT_DIR to /usr/local. On macOS, OpenSSL *\n"
```

Comment:
Thank you for the changes. I'll appreciate it if you could adjust the word wrapping here.

Reply:
Fixed! 5d73fcf86d676def7f9555d23d00996a9e3fc91e

Reply:
Oops. Forgot to push my commit. Pushed now.